go/vt/vtgate/vstream_manager.go

```diff
@@ -411,18 +411,54 @@ func (vs *vstream) alignStreams(ctx context.Context, event *binlogdatapb.VEvent,
 	}
 }
 
-func (vs *vstream) getCells() []string {
+// getCells determines the availability zones to select tablets from.
+// 3 scenarios:
+//
+// 1. No cells specified by the client via the gRPC request and vstreamCellAliasFallback = False
+// Tablets from the local cell of the VTGate will be selected by default
+//
+// 2. No cells specified by the client via the gRPC request and vstreamCellAliasFallback = True
+// Tablets from the local cell AND the cell alias that the VTGate's local cell belongs to qill be selected/
+// The local cell of the VTGate will take precendence over any other cell in the alias.
+//
+// 3. Cells are specified by the client via the gRPC request
+// These cell will take precendence over vstreamCellAliasFallback and only tablets belonging to the specified cells will be selected.
+// If the "local" tag is passed in as an option in the list of optCells,
+// the local cell of the VTGate will take precedence over any other cell specified.
+func (vs *vstream) getCells(ctx context.Context) []string {
 	var cells []string
 	if vs.optCells != "" {
-		for _, cell := range strings.Split(strings.TrimSpace(vs.optCells), ",") {
+		for i, cell := range strings.Split(strings.TrimSpace(vs.optCells), ",") {
+			// if the local tag is passed in, we must give local cell priority
+			// during tablet selection. Append the VTGate's local cell to the list of cells
+			if i == 0 && cell == "local" {
```

Comment:
Should we use `local:` instead of `local` here? This way it is uniform with the way we request cell locality. 

Also, if a user names a cell as `local` then it doesn't cause any ambiguity.

Reply:
It seems a bit strange to have a semicolon without anything to the right of it, but yeah we do need some differentiator if someone can also name a cell "local"